cmd/ndm_daemonset/controller/blockdevicestore.go

```diff
@@ -68,67 +71,49 @@ func (c *Controller) CreateBlockDevice(blockDevice apis.BlockDevice) error {
 		return err
 	}
 
-	/*
-	 * Update might failed due to to resource version mismatch which
-	 * can happen if some other entity updating same resource in parallel.
-	 */
-	err = c.UpdateBlockDevice(blockDevice, nil)
-	if err == nil {
-		return err
-	}
 	klog.Error("Update to blockdevice object failed: ", blockDevice.ObjectMeta.Name)
 	return nil
 }
 
+func boolPtr(b bool) *bool {
+	return &b
+}
+
 // UpdateBlockDevice update the BlockDevice resource in etcd
-func (c *Controller) UpdateBlockDevice(blockDevice apis.BlockDevice, oldBlockDevice *apis.BlockDevice) error {
+func (c *Controller) UpdateBlockDevice(blockDevice apis.BlockDevice) error {
 	var err error
 
-	blockDeviceCopy := blockDevice.DeepCopy()
-	if oldBlockDevice == nil {
-		oldBlockDevice = blockDevice.DeepCopy()
-		err = c.Clientset.Get(context.TODO(), client.ObjectKey{
-			Namespace: oldBlockDevice.Namespace,
-			Name:      oldBlockDevice.Name}, oldBlockDevice)
-		if err != nil {
-			klog.Errorf("eventcode=%s msg=%s : %v, err:%v rname=%v",
-				"ndm.blockdevice.update.failure",
-				"Failed to update block device : unable to get blockdevice object",
-				oldBlockDevice.ObjectMeta.Name, err, blockDeviceCopy.ObjectMeta.Name)
-			return err
-		}
-	}
-
-	blockDeviceCopy = mergeBlockDeviceData(*blockDeviceCopy, *oldBlockDevice)
```

Comment:
This function needs to be used for merging as we dont need to merge all the fields. Some fields are merged on condition depending on the status of the blockdevice.